### PR TITLE
buildtest command line tutorial

### PR DESCRIPTION
**Pull request by shahzebsiddiqui**
This PR is going to add the buildtest command line tutorial  The page is rendered see https://buildtest--1353.org.readthedocs.build/en/1353/command_line_tutorial.html

@prathmesh4321 @Xiangs18 @jscook2345 can you please review and provide feedback.

### CODE REVIEW

File: docs/command_line_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+    .. command-output:: buildtest build -t python
+       :shell:
+
+To learn more about how to run tests, please refer to :ref:`building_test` guide.
+
+Querying Test Report
+-----------------------
+
+buildtest will keep track of all tests in a report file (JSON) that can be used to display content of all test runs
```
Comment (prathmesh4321) [inline docs/command_line_tutorial.rst]: 'b' in buildtest should be capital 'B'.

File: docs/command_line_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+Building Test
+---------------
+
+The ``buildtest build`` command is used for running a test on your system given a :ref:`buildspec <what_is_buildspec>`
+file (YAML). The most common way to build a test is specifying a file path via ``buildtest build -b <path>``. To get started,
+let's build our first test by running the following::
+
+    buildtest build -b $BUILDTEST_ROOT/tutorials/hello_world.yml
+
+If you ran this successfully, you should see the following output
```
Comment (prathmesh4321) [inline docs/command_line_tutorial.rst]: Lines 22-24 can be removed as the command is already shown and executed in dropdown below.

File: docs/command_line_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    .. command-output:: buildtest build -b $BUILDTEST_ROOT/tutorials/hello_world.yml -b $BUILDTEST_ROOT/general_tests/configuration
+       :shell:
+
+The ``-x`` option can be used to exclude buildspecs which works similar to ``-b`` where it can be a file or directory path.
+In the next example try running the following commands::
+
+    buildtest build -b general_tests/configuration -x general_tests/configuration/ulimits.yml
+    buildtest build -b general_tests/configuration -x general_tests/configuration
+
+You will notice in the 2nd command when buildtest has no buildspecs to build it will terminate immediately.
```
Comment (prathmesh4321) [inline docs/command_line_tutorial.rst]: can use "second" instead of 2nd

File: docs/command_line_tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+    buildtest path hello_world
+    buildtest path -o hello_world
+    buildtest path -e hello_world
+
+We encourage you review :ref:`test_reports` for a detailed guide on how to query test in buildtest.
+
+Interacting with Buildspecs
+----------------------------
+
+buildtest supports several ways to interact with buildspecs, such as querying buildspec cache,
```
Comment (prathmesh4321) [inline docs/command_line_tutorial.rst]: 'b' in buildtest should be capital 'B'.

File: docs/command_line_tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+    buildtest rt --pass
+
+Now let's assume you want to know total failed tests in report file, you can use ``--row-count`` option which
+displays total row count. Let's run the following and see total fail count::
+
+    buildtest rt --fail --row-count
+
+The ``buildtest rt summary`` can be useful if you want to summary of report file.
+
+buildtest supports paging support with ``buildtest rt`` which can be useful when you
```
Comment (prathmesh4321) [inline docs/command_line_tutorial.rst]: 'b' in buildtest should be capital 'B'.

**Review by prathmesh4321 — CHANGES_REQUESTED** [review]
@shahzebsiddiqui I think we should have a dropdown for other commands mentioned in this tutorial as well just so that it looks consistent.

File: docs/command_line_tutorial.rst (hunk 1 above)
Comment (shahzebsiddiqui) [reply docs/command_line_tutorial.rst]: fixed

File: docs/command_line_tutorial.rst (hunk 3 above)
Comment (shahzebsiddiqui) [reply docs/command_line_tutorial.rst]: fixed

File: docs/command_line_tutorial.rst (hunk 4 above)
Comment (shahzebsiddiqui) [reply docs/command_line_tutorial.rst]: fixed

File: docs/command_line_tutorial.rst (hunk 5 above)
Comment (shahzebsiddiqui) [reply docs/command_line_tutorial.rst]: fixed

File: docs/command_line_tutorial.rst (hunk 2 above)
Comment (shahzebsiddiqui) [reply docs/command_line_tutorial.rst]: fixed

**Comment by shahzebsiddiqui** [comment]
> @shahzebsiddiqui I think we should have a dropdown for other commands mentioned in this tutorial as well just so that it looks consistent.

Do you mean for every command. The drop-down was there for some commands followed by `command-output` which will actually run the command and show the output. I didnt want to run every command and show output in the tutorial since that would cause every command to be run during documentation build and the purpose of the tutorial was to have user run the command with a simple copy-paste for the most part.


> @shahzebsiddiqui I think we should have a dropdown for other commands mentioned in this tutorial as well just so that it looks consistent.



> @shahzebsiddiqui I think we should have a dropdown for other commands mentioned in this tutorial as well just so that it looks consistent.

**Comment by prathmesh4321** [comment]
> > @shahzebsiddiqui I think we should have a dropdown for other commands mentioned in this tutorial as well just so that it looks consistent.
> 
> Do you mean for every command. The drop-down was there for some commands followed by `command-output` which will actually run the command and show the output. I didnt want to run every command and show output in the tutorial since that would cause every command to be run during documentation build and the purpose of the tutorial was to have user run the command with a simple copy-paste for the most part.
> 
> > @shahzebsiddiqui I think we should have a dropdown for other commands mentioned in this tutorial as well just so that it looks consistent.
> 
> > @shahzebsiddiqui I think we should have a dropdown for other commands mentioned in this tutorial as well just so that it looks consistent.

Okay that makes sense got it !

**Comment by shahzebsiddiqui** [comment]
@prathmesh4321 @Xiangs18 @jscook2345 do you think this content looks okay? i'd like to have this available as part of the buildtest 1.2 release scheduled for Feb 6th

**Pull request merged**